sherpa/fit.py

```diff
@@ -1403,7 +1429,7 @@ def report_progress(i, lower, upper):
             # If maximum number of refits has occurred, don't
             # try to reminimize again.
             if (hasattr(self.estmethod, "maxfits") and
-                    not(self.refits < self.estmethod.maxfits - 1)):
+                    not self.refits < (self.estmethod.maxfits - 1)):
```

Comment:
This is technically true, because arithmetic comparison has precedence over logical NOT. However, I had to look that up, because I've definitely done things like "False < 3" accidentally and that also works in Pytohn, leading to hard-to-find bugs.

Just to help future readers, let keep the outer parenthesis in addition to the inner ones that are added here as a bugfix:

```suggestion
                    not (self.refits < (self.estmethod.maxfits - 1))):
```

Reply:
Yes - there's a delicate dance to be had between accepting a linting suggestion to remove brackets/precedence/logic and keeping things because they more-clearly express intent!